test/Microsoft.Health.Fhir.Shared.Tests.E2E/Rest/HistoryTests.cs

```diff
@@ -150,19 +153,21 @@ public async Task GivenAValueForSinceAndBeforeCloseToLastModifiedTime_WhenGettin
 
             // First make a few edits
             _createdResource.Resource.Text = new Narrative { Div = "<div>Changed by E2E test</div>" };
-            await _client.UpdateAsync<Observation>(_createdResource);
-            newResources.Add(await _client.CreateAsync(Samples.GetDefaultPatient().ToPoco()));
-            newResources.Add(await _client.CreateAsync(Samples.GetDefaultOrganization().ToPoco()));
+            var tag = new Coding(string.Empty, Guid.NewGuid().ToString());
+            await _client.UpdateAsync<Observation>(AddTagToResource(_createdResource.Resource, tag));
+
+            newResources.Add(await _client.CreateAsync(AddTagToResource(Samples.GetDefaultPatient().ToPoco(), tag)));
+            newResources.Add(await _client.CreateAsync(AddTagToResource(Samples.GetDefaultOrganization().ToPoco(), tag)));
             Thread.Sleep(1000);
-            newResources.Add(await _client.CreateAsync(Samples.GetJsonSample("BloodGlucose").ToPoco()));
-            newResources.Add(await _client.CreateAsync(Samples.GetJsonSample("BloodPressure").ToPoco()));
-            newResources.Add(await _client.CreateAsync(Samples.GetJsonSample("Patient-f001").ToPoco()));
-            newResources.Add(await _client.CreateAsync(Samples.GetJsonSample("Condition-For-Patient-f001").ToPoco()));
+            newResources.Add(await _client.CreateAsync(AddTagToResource(Samples.GetJsonSample("BloodGlucose").ToPoco(), tag)));
+            newResources.Add(await _client.CreateAsync(AddTagToResource(Samples.GetJsonSample("BloodPressure").ToPoco(), tag)));
+            newResources.Add(await _client.CreateAsync(AddTagToResource(Samples.GetJsonSample("Patient-f001").ToPoco(), tag)));
+            newResources.Add(await _client.CreateAsync(AddTagToResource(Samples.GetJsonSample("Condition-For-Patient-f001").ToPoco(), tag)));
```

Comment:
Probably this can be simplified(just for readability) a little by creating a method like below
`private async Task<Resource> CreateResourceWithTag(Resource resource, Coding tag)
        {
            return await _client.CreateAsync(AddTagToResource(resource, tag));
        }`

Reply:
Agreed.  Updated.